src/main/java/com/lorepo/icplayer/client/module/ordering/OrderingPrintable.java

```diff
@@ -1,93 +1,256 @@
+     *                       String type. Numbers represent indexes of items. The numbers are arranged in order.
+     *                       Numbers must be separated by a comma. Example: "3,1,2"
+     * @return indexes items in order from the state
+     */
+    private Integer[] getItemsIndexesInOrderFromState(
+            HashMap<String, String> printableState) {
+        String orderState = printableState.get("order");
+        String[] indexes = orderState.split(",");
+        Integer[] parsedIndexes = new Integer[indexes.length];
+        for (int i = 0; i < indexes.length; i++) {
+            parsedIndexes[i] = Integer.parseInt(indexes[i]);
+        }
+        return parsedIndexes;
+    }
+
+    private Element createPrintableIndexBoxIfPrintableState(
+            Integer itemPosition, boolean checkAnswers,
+            boolean isCorrectOrder) {
+        if (checkAnswers) {
+			return createPrintableIndexBoxForCheckAnswers(
+					itemPosition, isCorrectOrder);
 		}
-		
-		for (int i = 0; i < parsedItems.length; i++) {
-			result += parsedItems[i];
+        return createPrintableIndexBoxForShowUserAnswers(itemPosition);
+    }
+
+    private Element createPrintableIndexBoxForCheckAnswers(
+            Integer itemPosition, boolean isCorrectOrder) {
+        if (isCorrectOrder) {
+			return createPrintableIndexBoxWithCorrectSign(itemPosition);
 		}
-		
-		result += "</div>";
-		result = PrintableContentParser.addClassToPrintableModule(result, className, !model.isSplitInPrintBlocked());
-		return result;
-	}
-	
-	private String createItem(OrderingItem item, boolean isVertical, int index) {
-		// index == 0 means no index will be displayed
-		String result = "";
-		String displayMode = isVertical? "block" : "inline-block";
-		result += "<div style=\"display:" + displayMode + "\">";
-		result += "<div class=\"item-wrapper\">";
-		result += "<table><tr><td>";
-		result += "<div class=\"number-box\">";
-		if (index != 0) {
-		    result += Integer.toString(index);
+        return createPrintableIndexBoxWithWrongSign(itemPosition);
+    }
+
+    private Element createPrintableIndexBoxForShowUserAnswers(
+            Integer itemPosition) {
+        return createPrintableIndexBox(itemPosition);
+    }
+
+    private Element createPrintableIndexBoxIfNotPrintableState(
+            Integer itemPosition, boolean showAnswers) {
+        if (showAnswers) {
+			return createPrintableIndexBoxForShowAnswers(itemPosition);
 		}
-		result += "</div>";
-		result += "</td><td>";
-		result += item.getText();
-		result += "</td></tr></table></div></div>";
-		return result;
-	}
-	
-	private int nextInt(int upperBound) {
-		int result = 0;
-		if (controller != null) {
-			result = controller.nextInt(upperBound);
-		} else {
-			result = Random.nextInt(upperBound);
+        return createPrintableIndexBoxForEmptyState();
+    }
+
+    private Element createPrintableIndexBoxForShowAnswers(

```

Comment:
W jakim celu są funkcje, które jedyne co robią to zwracają wywołanie z innej funkcji?